### PR TITLE
add optional serde support

### PR DESCRIPTION
**Pull request by zhengyi-yang**
#40

### CODE REVIEW

**Comment by jrraymond** [comment]
Can you add add some tests?

File: Cargo.toml [hunk 1]
```diff
@@ -18,3 +18,6 @@ default = ["std"]
 [package.metadata.release]
 no-dev-version = true
 tag-name = "{{version}}"
+
+[dependencies]
+serde = { version = "1.0", features = ["derive"], optional = true }
```
Comment (jrraymond) [inline Cargo.toml:23]: Can you add a trailing new line, some linux tools (ie `wc`) expect the last line to end in a new line.

Comment (zhengyi-yang) [reply Cargo.toml]: Thanks, I've added a new line

**Comment by zhengyi-yang** [comment]
@jrraymond I wonder if it's necessary to add tests, because this totally relies on the automatic implementation of `serde`. Since `FixedBitSet` only contains a `Vec<u32>` and a `usize`, it should be okay as long as the crate `serde` has been properly tested.

**Comment by jrraymond** [comment]
I think it's worth it to add a few tests because they are relatively easy to add. 

The tests in `serde` verify that `Vec<u32>` and `usize` serialize/deserialize correctly. A test would verify a `FixedBitSet` `X` serializes to bytes `Y`.  This test has value because if the layout of `FixedBitSet` changes, for example the block size changes to `u64`, then `FixedBitSet` `X` might serialize to bytes `Y'`. This might be acceptable, but a test would make such an API change obvious.

**Comment by zhengyi-yang** [comment]
No problem.

**Pull request closed without merging**